lib/jekyll/document.rb

```diff
@@ -262,12 +270,16 @@ def published?
     def read(opts = {})
       Jekyll.logger.debug "Reading:", relative_path
 
-      if yaml_file?
+      if yaml_file? && !render_yaml_file?
         @data = SafeYAML.load_file(path)
       else
         begin
-          merge_defaults
-          read_content(opts)
+          merge_defaults 
+          if yaml_file?
+            read_yml_content()
```

Comment:
**Tip:** This line isn't going to be evaluated because `ln 275` above has already established that the current document is not a YAML File

Reply:
This does hit as the if statement is evaluating if it is a yaml file but not to be written. We then only get into that else if the yaml file should be written or its not a yaml file at all.